index.html

```diff
@@ -1615,6 +1618,211 @@ <h3>
+          {
+            "id": "https://example.com/app",
+            "name": "My App",
+            "icons": [{
+              "src": "icon/hd_hi",
+              "sizes": "128x128"
+            }],
+            "start_url": "/app/index.html",
+            "scope": "/app",
+            "display": "standalone",
+            "scope_extensions": [
+              { "type": "origin", "origin": "https://example.co.uk" },
+              { "type": "origin", "origin": "https://help.example.com" }]
+          }
+        </pre>
+        <p>
+          The following shows 2 [=web-app-origin-association=] files which can
+          be downloaded from the `.well-known` path of the [=origins=]  listed 
+          in the `scope_extensions` member.
+        </p>
+        <pre class="example json" 
+        title="https://example.co.uk/.well-known/web-app-origin-association">
+          {
+            "https://example.com/app": {
+              "scope": "/app"   
+            }
+          }
+        </pre>
+        <pre class="example json" 
+        title="https://help.example.com/.well-known/web-app-origin-association">
+          {
+            "https://example.com/app": {
+              "scope": "/"   
+            }
+          }
+        </pre>
+        <p> 
+          The navigation scope of this app consists of URLs that are 
+          [=URL/within scope=] of any of these [=URLs=]:
+          `https://example.com/app`, `https://example.co.uk/app`, and 
+          `https://help.example.com`. 
+        </p>
+      </section>
+      <section>
+      <h2>Processing the scope_extensions member</h2>
+      <p>
+        To <dfn>process the `scope_extensions` member</dfn>, given
+        [=ordered map=] |json:ordered map| and [=ordered map=]
+        |manifest:ordered map|:
+      </p>
+      <ol class="algorithm">
+        <li>Let |processedScopeExtensions:list| be a new [=list=].
+        </li>
+        <li>Set |manifest|["scope_extensions"] to |processedScopeExtensions|.
+        </li>
+        <li>If |json|["scope_extensions"] doesn't [=map/exist=] or is not a 
+          [=list=], return.
+        </li>
+        <li>[=list/For each=] |entry:ordered map| of |json|["scope_extensions"]:
+          <ol>
+            <li>If |entry|["type"] or |entry|["origin"] is undefined,
+              [=iteration/continue=].
+            </li>
+            <li>If |entry|["type"] is not "origin", [=iteration/continue=].
+            </li>
+            <li>If |entry|["origin"] is not a [=string=],
+              [=iteration/continue=].
+            </li>
+            <li>Set |entry|["origin"] to the result of parsing it as an
+              [=origin=].
+            </li>
+            <li>Let [=URL=] |resolved_scope:URL| be an empty [=URL=]. </li>
+            <li>Validate |entry| using the algorithm from [=validate scope
+              extensions=], given [=ordered map=] |entry:ordered map|, [=URL=]
+              |manifest_id:URL|, and [=URL=] |resolved_scope:URL|. 
```

Comment:
why do we pass it the scope? Can it return that?

Looking at the manifest spec, we can say that things 'return an error', and we can check for that instead of 'validation'.

Reply:
Changed to return error or resolved scope URL.